src/main/java/com/flightstats/hub/spoke/SpokeContentDao.java

```diff
@@ -58,4 +74,28 @@
             throw e;
         }
     }
+
+    Optional<ChannelContentKey> getOldestItem(SpokeStore store) {
+        String storePath = HubProperties.getSpokePath(store);
+        logger.trace("getting oldest item from " + storePath);
+        // expected result format: YYYY-MM-DD+HH:MM:SS.SSSSSSSSSS /mnt/spoke/store/channel/yyyy/mm/dd/hh/mm/ssSSShash
+        String command = String.format(GET_OLDEST_ITEM_COMMAND, storePath);
+        String result = StringUtils.chomp(commander.runInBash(command, 3));
+        if (StringUtils.isEmpty(result)) return Optional.absent();
+        String spokePath = StringUtils.substring(result, 31);
+        try {
+            return Optional.of(ChannelContentKey.fromSpokePath(spokePath));
+        } catch (IllegalArgumentException e) {
```

Comment:
The Optional here is used because we're running `find` in a Bash shell and then executing logic on the result. Theres a chance the command times out or returns unexpected values.